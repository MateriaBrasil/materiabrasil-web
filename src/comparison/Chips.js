import React from 'react'
import Chip from '@material-ui/core/Chip'
import Grid from '@material-ui/core/Grid'
import map from 'lodash/map'

import removeItem from './removeItem'
import Colors from './Colors'

export default ({ list, alignItems, remove }) => (
  <Grid
    item
    xs={12}
    lg={3}
    md={3}
    style={{
      display: 'flex',
      flexDirection: 'column',
      alignItems,
      justifyContent: 'center',
    }}
  >
    {map(list, ({ id, name }, index) => (
      <Chip
        key={id}
        label={name}
        style={{ background: Colors[index], marginBottom: 10 }}
        onDelete={removeItem(id, remove)}
        color="secondary"
      />
    ))}
  </Grid>
)
