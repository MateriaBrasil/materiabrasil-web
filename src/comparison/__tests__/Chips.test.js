import React from 'react'
import renderer from 'react-test-renderer'

import Chips from '../Chips'

jest.mock('@material-ui/core/Grid', () => props => (
  <div {...props}>Grid - {props.children}</div>
))

jest.mock('@material-ui/core/Chip', () => props => (
  <div {...props}>Chip - {props.children}</div>
))

const props = {
  list: [{ id: 1, name: 'foo' }],
  remove: jest.fn(),
}

it('renders correctly', () => {
  const tree = renderer.create(<Chips {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
