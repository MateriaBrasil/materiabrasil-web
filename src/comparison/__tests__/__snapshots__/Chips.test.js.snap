// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  item={true}
  lg={3}
  md={3}
  style={
    Object {
      "alignItems": undefined,
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
  xs={12}
>
  Grid - 
  <div
    color="secondary"
    label="foo"
    onDelete={[Function]}
    style={
      Object {
        "background": "#239eb1",
        "marginBottom": 10,
      }
    }
  >
    Chip - 
  </div>
</div>
`;
